packages/dds/sequence/src/revertibles.ts

```diff
@@ -374,12 +382,23 @@ function revertLocalDelete(
 	revertible: TypedRevertible<typeof IntervalOpType.DELETE>,
 ) {
 	const label = revertible.interval.properties.referenceRangeLabels[0];
+	const collection = string.getIntervalCollection(label);
 	const start = string.localReferencePositionToPosition(revertible.start);
+	const startSlide = collection.getSlideToSegment(revertible.start);
+	const startSlidePos: number =
+		startSlide !== undefined && string.getPosition(startSlide.segment) !== -1
+			? string.getPosition(startSlide.segment) + (startSlide.offset as number)
```

Comment:
we should try to avoid `as` casting as much as we can. it is unsafe. you are basically telling the compiler you know best, and to do this regardless of what it thinks. if the type were to change this would not be caught until runtime.